doc/neps/nep-0018-array-function-protocol.rst

```diff
@@ -310,31 +395,32 @@ between these two dispatch protocols, we should
 `also update <https://github.com/numpy/numpy/issues/11306>`_
 ``__array_ufunc__`` to match this behavior.
 
-Special handling of ``numpy.ndarray``
-'''''''''''''''''''''''''''''''''''''
+The ``__array_function__`` method on ``numpy.ndarray``
+''''''''''''''''''''''''''''''''''''''''''''''''''''''
 
 The use cases for subclasses with ``__array_function__`` are the same as those
-with ``__array_ufunc__``, so ``numpy.ndarray`` should also define a
-``__array_function__`` method mirroring ``ndarray.__array_ufunc__``:
+with ``__array_ufunc__``, so ``numpy.ndarray`` also defines a
+``__array_function__`` method.
+
+``ndarray.__array_function__`` is a trivial case of the "Defaulting to NumPy's
+implementation" strategy described above: *every* NumPy function on NumPy
+arrays is defined by calling NumPy's own implementation if there are other
+overrides:
 
 .. code:: python
 
     def __array_function__(self, func, types, args, kwargs):
-        # Cannot handle items that have __array_function__ other than our own.
-        for t in types:
-            if (hasattr(t, '__array_function__') and
-                    t.__array_function__ is not ndarray.__array_function__):
-                return NotImplemented
+        if not all(issubclass(t, ndarray) for t in types):
```

Comment:
cc @mhvk I still think this should be changed to

```suggestion
        if hasattr(t, '__array_function__') and t.__array_function__ is not ndarray.__array_function__:
```
Could you think of a good reason why this shouldn't be the case, especially since `__skip_array_function__` is now agreed upon.

Reply:
This was an intentional change, see https://github.com/numpy/numpy/pull/12328. Subclass implementations get used because the protocol for calling `__array_function__` methods calls subclasses first.

I've added a paragraph to explain, please take a look.

Reply:
The new paragraph looks good, thanks.